packages/web3-eth/src/rpc_method_wrappers.ts

```diff
@@ -1177,17 +1186,19 @@ export function sendTransaction<
 								) as unknown as ResolveType,
 							);
 						} else if (transactionReceipt.status === BigInt(0)) {
+							const error = await getTransactionError<ReturnFormat>(
+								web3Context,
+								transactionFormatted as TransactionCall,
+								transactionReceiptFormatted,
+								undefined,
+								options?.contractAbi,
+							);
+
 							if (promiEvent.listenerCount('error') > 0) {
-								promiEvent.emit(
-									'error',
-									new TransactionError(
-										'Transaction failed',
-										transactionReceiptFormatted,
-									),
-								);
+								promiEvent.emit('error', error);
 							}
-							reject(transactionReceiptFormatted as unknown as ResolveType);
-							return;
+
+							reject(error);
```

Comment:
This will result in inconsistent function behavior. If transaction status is even 0x0 and tx receipt is **available** we should reject with receipt obj and only emit error. as 0x0 can transaction is **mined** but failed.

Reply:
@jdevcs Rejecting with only the receipt provides no context to the user as to why their transaction failed (or even that it failed if the user doesn't know to check for the `status`). Additionally, users may not be using an event listener for `error` events, so we shouldn't force that as the only option to obtain why the transaction failed. `1.x` also doesn't return the receipt for failed transaction, it throws an error like so:

```typescript
Error: Returned error: execution reverted: This is a send revert
    at Object.ErrorResponse (node_modules/web3-core-helpers/lib/errors.js:28:19)
    at node_modules/web3-core-requestmanager/lib/index.js:300:36
    at node_modules/web3-providers-http/lib/index.js:127:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  data: '0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000155468697320697320612073656e64207265766572740000000000000000000000'
}
```

Reply:
Thanks Wyatt for this discussion points. 

> or even that it failed if the user doesn't know to check for the status

 >users may not be using an event listener for error events
 
 
 I think above mentioned are few assumptions and we dnt have any user survey data for these?
 
How will user get tx hash in case of reject-error? : via `receipt` event listener may be. That falls under:
  >users may not be using an event listener for <del>error</del> events
 
 When a tx is mined (fails or succeeds), the most primary thing to Directly point is **tx hash** ( by default we can return it via receipt which also has status flag), and it can be used for making any number of further queries later explicitly. we have opportunity here for improvements as compared to 1.x due to major bump.
 
 If tx is not mined we can throw error.

Reply:
@jdevcs A brand new user to ETH that's creating their first Dapp is most likely not going to know all the intricacies of an ETH transaction, and I think explicitly erroring with an `Error` is more helpful than returning the transaction receipt the same way we'd return for a successful transaction

For all the errors that `getTransactionError` can return for failed mined transactions, the receipt is a property on the error object and can be accessed if the error is caught

Reply:
> For all the errors that getTransactionError can return for failed mined transactions, the receipt is a property on the error object and can be accessed if the error is caught

that can be helpful.